examples/render/src/main.rs

```diff
@@ -67,17 +66,136 @@ struct Args {
+            DebugMode::DepthBuffer => MapDebugOptions::DepthBuffer,
+        }
+    }
 }
+
+impl Args{
+    fn render(self)->Vec<u8>{
+        let mut map = ImageRendererOptions::new();
+        map.with_api_key(self.apikey.unwrap_or_default());
+        map.with_cache_path(self.cache.to_string_lossy().to_string());
+        map.with_asset_root(self.asset_root.to_string_lossy().to_string());
+        map.with_pixel_ratio(self.ratio);
+        map.with_size(self.width, self.height);
+    
+        match self.mode {
+            Mode::Static => {
+                let mut map = map.build_static_renderer();
+                if let Some(debug) = self.debug {
+                    map.set_debug_flags(debug.into());
+                }
+                map.set_style_url(&self.style);
+                map.set_camera(
+                    self.x as f64,
+                    self.y as f64,
+                    self.zoom as f64,
+                    self.bearing,
+                    self.pitch,
+                );
+                map.render_static().as_slice().to_vec()
+            }
+            Mode::Tile => {
+                if self.bearing != 0.0 || self.pitch != 0.0 {
+                    println!(
+                        "Warning: bearing and pitch are not supported in tile-mode and will be ignored"
+                    );
+                }
+                let mut map = map.build_tile_renderer();
+                map.set_style_url(&self.style);
+                if let Some(debug) = self.debug {
+                    map.set_debug_flags(debug.into());
+                }
+                map.render_tile(self.zoom, self.x, self.y)
+                    .as_slice()
+                    .to_vec()
+            }
+            Mode::Continuous => {
+                todo!("not yet implemented in the wrapper")
+            }
+        }
+    }
+}
+
+fn main() {
+    let args = Args::parse();
+    println!("Rendering arguments: {:#?}", args);
+    let output = args.output.clone();
+
+    let before_initalisation = Instant::now();
+    let data = args.render();
+    println!(
+        "Rendering successfull in {elapsed:?}, writing result to {output:?}",
+        elapsed = before_initalisation.elapsed()
+    );
+    println!("Note: Rendering subsequent tiles/images would be faster as initialization is amortized.");
+    fs::write(&output, &data)
+        .expect(&format!("Failed to write rendered map to {output:?}"));
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_rendering() {
+        let args = Args {
+            width: 32,
+            height: 32,
+            mode: Mode::Static,
+            ..Args::parse()
+        };
+        let data = args.render();
+        assert!(!data.is_empty());
+        
+        let args = Args {
+            width: 64,
+            height: 64,
+            mode: Mode::Tile,
+            ..Args::parse()
+        };
+        let data = args.render();
+        assert!(!data.is_empty());
+    }
```

Comment:
One instead of two testcases as otherwise this deadlocks.
Somewhere there is global state apparently.

Reply:
one item for sure - the sqlite cache is already a single point of contention, unless each thread is made independent

Reply:
My thought exactly, but that is not the cause.
If I set this to a different filename, it does not resolve the issue.

FYI: deadlocks supprisingly repeatably.. maybe something else is going on.